Mathlib/SetTheory/Cardinal/Basic.lean

```diff
@@ -937,6 +954,19 @@ theorem prod_eq_zero {ι} (f : ι → Cardinal.{u}) : prod f = 0 ↔ ∃ i, f i
 
 theorem prod_ne_zero {ι} (f : ι → Cardinal) : prod f ≠ 0 ↔ ∀ i, f i ≠ 0 := by simp [prod_eq_zero]
 
+theorem power_sum {ι} (a : Cardinal) (f : ι → Cardinal) :
+    a ^ sum f = prod fun i ↦ a ^ f i :=
+  inductionOn a fun α ↦ by
+    apply induction_on_pi f fun f ↦ ?_
+    rw [prod, sum]
+    simp_rw [power_def]
+    convert mk_sigma_arrow α f using 1
+    · exact mk_congr <| Equiv.arrowCongr
+        (Equiv.sigmaCongr (Equiv.cast rfl) (fun _ ↦ outMkEquiv)) (Equiv.refl α)
+    · apply mk_pi_congr
+      intro i
+      rw [mk_out]
```

Comment:
What do you think of this proof, which doesn't need any of the new `congr` lemmas?
```suggestion
    a ^ sum f = prod fun i ↦ a ^ f i := by
  induction a using Cardinal.inductionOn with | _ α => ?_
  induction f using induction_on_pi with | _ f =>
  simp_rw [prod, sum, power_def]
  apply mk_congr
  refine (Equiv.piCurry fun _ _ => α).trans ?_
  refine Equiv.piCongrRight fun b => ?_
  refine (Equiv.arrowCongr outMkEquiv (Equiv.refl α)).trans ?_
  exact (outMkEquiv).symm
```

I'd like to hear @b-mehta's thoughts on this before merging

Reply:
I still like the congruency lemmas since I always want to have more ways to reason about cardinals without reasoning about `Equiv`'s, but I'll remove them if you disagree. @vihdzp what do you think?

Reply:
I'm compelled to agree. You don't usually see explicit equivalences when doing cardinal arithmetic after a certain point, instead you just use results like Cantor-Schröder to do all the heavy lifting. So being able to abstract that away is a good thing in my book.

Reply:
I already commited the new proof, it's neat. I think I'll move all the congruency lemmas to #16824 to have all the versions in one place, now that they're not needed here.